Add additional debug logs when Connection::dispatchMessage fails

achristensen07 opened the pull request:
#### 47c248c8e3cda592cbec92eb839b9bd4b0d1e23b
<pre>
Add additional debug logs when Connection::dispatchMessage fails
<a href="https://bugs.webkit.org/show_bug.cgi?id=287151">https://bugs.webkit.org/show_bug.cgi?id=287151</a>
<a href="https://rdar.apple.com/144306094">rdar://144306094</a>

Reviewed by Sihui Liu.

Sometimes I hit the assertion in the end of the function, and I always
wish it would tell me which message decode failed.  This makes it so.

* Source/WebKit/Platform/IPC/Connection.cpp:
(IPC::Connection::dispatchMessage):

Canonical link: <a href="https://commits.webkit.org/290638@main">https://commits.webkit.org/290638@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f15ee5123eca572d212309dfe9ec688b306c8363

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/88527 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8046 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/42967 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/93485 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39281 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8433 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16230 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68272 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25983 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/91529 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6453 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/80067 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48634 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6224 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38389 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76595 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/35366 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95327 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/15702 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/11501 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77135 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/15958 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/75923 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76399 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/20795 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/19213 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/8741 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/14275 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/15718 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/21026 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/15459 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/18908 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17241 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebKit/Platform/IPC/Connection.cpp:
```diff
@@ -1447,7 +1447,12 @@ void Connection::dispatchMessage(UniqueRef<Decoder> message)
-    ASSERT(!didReceiveInvalidMessage);
+#if ASSERT_ENABLED
+    if (didReceiveInvalidMessage) {
+        WTFLogAlways("Received invalid message %s for destination %" PRIu64, description(message->messageName()).characters(), message->destinationID());
```
I think there is log in `didReceiveInvalidMessage` functions, like `NetworkConnectionToWebProcess::didReceiveInvalidMessage`, so maybe we could just move `ASSERT(!didReceiveInvalidMessage);` to after `m_client->didReceiveInvalidMessage(...)`

achristensen07 replied:
AuxiliaryProcessProxy calls RELEASE_LOG_FAULT which doesn't output to stderr like WTFLogAlways does

achristensen07 committed:
Add additional debug logs when Connection::dispatchMessage fails
https://bugs.webkit.org/show_bug.cgi?id=287151
rdar://144306094

Reviewed by Sihui Liu.

Sometimes I hit the assertion in the end of the function, and I always
wish it would tell me which message decode failed.  This makes it so.

* Source/WebKit/Platform/IPC/Connection.cpp:
(IPC::Connection::dispatchMessage):

Canonical link: https://commits.webkit.org/290638@main
